library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -461,4 +515,47 @@ internal class TransactionPayloadFragment :
         }
         return result
     }
+
+    private fun MutableList<TransactionPayloadItem>.getCollapsableOrDefault(): List<TransactionPayloadItem> {
+        val default = this
+
+        return if (viewModel.isUsingCollapsableJson) {
+            try {
+                mapToJsonElements()
+            } catch (t: JsonSyntaxException) {
+                Toast.makeText(context, t.message, Toast.LENGTH_LONG).show()
+                Logger.error(t.message ?: "Error when formatting json")
+                viewModel.toggleCollapsableJson()
+                default
+            }
+        } else {
+            default
+        }
+    }
+
+    private fun MutableList<TransactionPayloadItem>.mapToJsonElements(): List<TransactionPayloadItem> {
+        val bodyBuilder = StringBuilder()
+        val newList = arrayListOf<TransactionPayloadItem>()
+
+        forEach { item ->
+            when (item) {
+                is TransactionPayloadItem.BodyLineItem -> bodyBuilder.append(item.line)
+                is TransactionPayloadItem.HeaderItem,
+                is TransactionPayloadItem.ImageItem -> newList.add(item)
+
+                else -> Unit
+            }
+        }
```

Comment:
Not a super big fan of this approach where you re-add `HeaderItem` and `ImageItem`.
The only thing you need to do is to map `BodyLineItem` to `BodyCollapsibleLineItem` if the user clicked. Could you clean this up a bit?

Reply:
Also can you make this when exhaustive?